### DISCUSSION TITLE
How to use with proxy_pass?

### DISCUSSION THREAD

**Discussion by roysG** [nginx-proxy/nginx-proxy#1802] [Q&A]
Great plugin, 
But i still can׳t figure it out how to use proxy_pass

When user typing my address:
https://api.example.com
I need to forwarded my domain to https://app:3001/api

**Comment by subdavis** [comment]
What have you tried?  This should be possible:

https://github.com/jwilder/nginx-proxy#per-virtual_host

Seems like you just need to get set up normally with VIRTUAL_HOST and VIRTUAL_PORT then add the proxy_pass config in a per-virtual_host custom config file.

**Comment by Phorze** [comment]
Just to clarify this question from my perspective: 

Using a location directive in a custom per-virtual-host config file does not allow you to overwrite the default location directive, which always has a proxy_pass http://$VIRTUAL_HOST format. We want to be able to change it to http://$VIRTUAL_HOST/path/

Using a vhost.d config file with the _location suffix will allow you to ADD configuration to the default location directive. Adding a proxy_pass there won't work since you would end up with a duplicate proxy_pass.

Hope this clarifies the question.

**Comment by sbpro86** [comment]
> Just to clarify this question from my perspective:
> 
> Using a location directive in a custom per-virtual-host config file does not allow you to overwrite the default location directive, which always has a proxy_pass http://$VIRTUAL_HOST format. We want to be able to change it to http://$VIRTUAL_HOST/path/
> 
> Using a vhost.d config file with the _location suffix will allow you to ADD configuration to the default location directive. Adding a proxy_pass there won't work since you would end up with a duplicate proxy_pass.
> 
> Hope this clarifies the question.

Exactly. It seems that there no possibility to rewrite to a custom path instead of /.

**Comment by ketsapiwiq** [comment]
Seems related to #1286 and to the #1179 PR.

**Comment by Pryla** [comment]
I am looking for solution to this question. In my case I want to proxy a tracking script from plausible analytics. Should I create another nginx service where I can completely customize my .conf file or what is a good solution here?

**Comment by johndevs** [comment]
I am stuck on the same issue. I would like to proxy_pass to a sub-directory from the default location block but cannot find a way to get there.
